test_util/cluster_api.py

```diff
@@ -271,244 +273,29 @@ def get_url(self, node, path, port=None):
-            check_tasks_healthy = (not check_health or data['app']['tasksHealthy'] == app_definition['instances'])
-
-            if (check_tasks_running and check_tasks_healthy):
-                res = [Endpoint(t['host'], t['ports'][0], t['ipAddresses'][0]['ipAddress'])
-                       if len(t['ports']) is not 0
-                       else Endpoint(t['host'], 0, t['ipAddresses'][0]['ipAddress'])
-                       for t in data['app']['tasks']]
-                logging.info('Application deployed, running on {}'.format(res))
-                return res
-            elif (not check_tasks_running):
-                logging.info('Waiting for application to be deployed: '
-                             'Not all instances are running: {}'.format(repr(data)))
-                return None
-            elif (not check_tasks_healthy):
-                logging.info('Waiting for application to be deployed: '
-                             'Not all instances are healthy: {}'.format(repr(data)))
-                return None
-            else:
-                logging.info('Waiting for application to be deployed: {}'.format(repr(data)))
-                return None
-
-        try:
-            return _pool_for_marathon_app(app_definition['id'])
-        except retrying.RetryError:
-            raise Exception("Application deployment failed - operation was not "
-                            "completed in {} seconds.".format(timeout))
-
-    def destroy_marathon_app(self, app_name, timeout=120):
-        """Remove a marathon app
-
-        Abort the test if the removal was unsuccesful.
-
-        Args:
-            app_name: name of the applicatoin to remove
-            timeout: seconds to wait for destruction before failing test
-        """
-        @retrying.retry(wait_fixed=1000, stop_max_delay=timeout * 1000,
-                        retry_on_result=lambda ret: not ret,
-                        retry_on_exception=lambda x: False)
-        def _destroy_complete(deployment_id):
-            r = self.get('/marathon/v2/deployments', headers=self._marathon_req_headers())
-            assert r.ok
-
-            for deployment in r.json():
-                if deployment_id == deployment.get('id'):
-                    logging.info('Waiting for application to be destroyed')
-                    return False
-            logging.info('Application destroyed')
-            return True
-
-        r = self.delete('/marathon/v2/apps' + app_name, headers=self._marathon_req_headers())
-        assert r.ok
-
-        try:
-            _destroy_complete(r.json()['deploymentId'])
-        except retrying.RetryError:
-            raise Exception("Application destroy failed - operation was not "
-                            "completed in {} seconds.".format(timeout))
+    def get_client(self, path, user='default', default_headers={}):
+        return test_util.helpers.ApiClient(
+            cluster=self,
+            user=self._get_user(user),
```

Comment:
why do we resolve user here but not just pass in headers? Seems like this should either just take the user name or take the auth headers.